File: frontend/src/types/suggestion.ts
/**
 * TypeScript type definitions for ONLSuggest suggestion system.
 */

/**
 * A single suggestion returned by the backend.
 */
export interface Suggestion {
  /**
   * Full-text question suggestion (e.g., "Hoe vraag ik een parkeervergunning aan in Amsterdam?")
   */
  suggestion: string;

  /**
   * Confidence score (0.0 to 1.0)
   */
  confidence: number;

  /**
   * Optional service information if matched
   */
  service?: {
    id: number;
    name: string;
    description: string;
    category: string;
  };

  /**
   * Optional gemeente information if matched
   */
  gemeente?: {
    id: number;
    name: string;
    description: string;
  };
}

/**
 * Response from the /api/v1/suggestions endpoint.
 */
export interface SuggestionResponse {
  /**
   * User's partial query input
   */
  query: string;

  /**
   * List of suggestions (max 5)
   */
  suggestions: Suggestion[];

  /**
   * Response time in milliseconds
   */
  response_time_ms: number;
}

File: project1/backend/app/schemas/suggestion.py
"""
Pydantic schemas for suggestions API
"""
from typing import List, Optional
from pydantic import BaseModel, Field


class SuggestionRequest(BaseModel):
    """Request schema for suggestions endpoint"""
    query: str = Field(..., min_length=2, description="Search query (minimum 2 characters)")
    max_results: int = Field(5, ge=1, le=10, description="Maximum number of suggestions")


class ServiceInfo(BaseModel):
    """Service information in suggestion"""
    id: Optional[int] = None
    name: str
    description: str
    category: str


class Suggestion(BaseModel):
    """Single suggestion response"""
    suggestion: str
    confidence: float
    service: ServiceInfo
    gemeente: Optional[str] = None


class SuggestionResponse(BaseModel):
    """Response schema for suggestions endpoint"""
    query: str
    suggestions: List[Suggestion]
    response_time_ms: float
    using_database: bool
    suggestion_engine: str
    fallback_occurred: Optional[bool] = None
    fallback_reason: Optional[str] = None
